tests/checks/methodology/identifier_leakage_test.py

```diff
@@ -29,13 +29,33 @@ def util_generate_dataframe_and_expected():
 
 
 def test_assert_identifier_leakage():
-    df, expected = util_generate_dataframe_and_expected()
+    df, expected = generate_dataframe_and_expected()
     result = IdentifierLeakage().run(dataset=Dataset(df, label='label', datetime_name='x2', index_name='x3'))
     for key, value in result.value.items():
         assert_that(key, is_in(expected.keys()))
         assert_that(value, close_to(expected[key], 0.1))
 
 
+def test_identifier_leakage_with_extracted_from_dataframe_index():
+    df, expected = generate_dataframe_and_expected()
+    df.set_index('x3', inplace=True)
+    dataset = Dataset(df=df, label='label', set_index_from_dataframe_index=True)
+    result = IdentifierLeakage().run(dataset=dataset)
+    for key, value in result.value.items():
+        assert_that(key, is_in(expected.keys()))
+        assert_that(value, close_to(expected[key], 0.1))
+
+
+def test_identifier_leakage_with_extracted_from_dataframe_datatime_index():
+    df, expected = generate_dataframe_and_expected()
+    df.set_index('x2', inplace=True)
+    dataset = Dataset(df=df, label='label', set_datetime_from_dataframe_index=True)
+    result = IdentifierLeakage().run(dataset=dataset)
+    for key, value in result.value.items():
+        assert_that(key, is_in(expected.keys()))
```

Comment:
Does this ensure that the name of the datetime column identified by IdentifierLeakage is indeed x2?